pkg/domain/infra/abi/generate.go

```diff
@@ -113,6 +113,9 @@ func (ic *ContainerEngine) GenerateKube(ctx context.Context, nameOrIDs []string,
 	if options.Replicas > 1 && options.Type != define.K8sKindDeployment {
 		return nil, fmt.Errorf("--replicas can only be set when --type is set to deployment")
 	}
+	if options.Replicas > 1 && options.Type == define.K8sKindDaemonSet {
```

Comment:
This check shouldn't be needed, the check above on line 113 should already handle this.